File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
number = {
    "zero": "0", "one": "1", "two": "2", "three": "3", "four": "4",
    "five": "5", "six": "6", "seven": "7", "eight": "8", "nine": "9"
    }

def solution(s):
    answer = s
    for k, v in number.items():
        answer = answer.replace(k, v)
    return int(answer)

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
import java.util.*;

class Solution {
    public int solution(String s) {
        Map<String, String> map = new HashMap<>();
        map.put("one", "1"); map.put("two", "2"); map.put("three", "3");
        map.put("four", "4"); map.put("five", "5"); map.put("six", "6"); 
        map.put("seven", "7"); map.put("eight", "8"); map.put("nine", "9"); map.put("zero", "0"); 

        
        String answer = s;
        for (Map.Entry<String, String> entry : map.entrySet()) {
            answer = answer.replace(entry.getKey(), entry.getValue());
        } 
        return Integer.parseInt(answer);
    }
}